### PR TITLE
[DESIGN2-216-Typography] Typography Component - DSCO - Update color variables

### PR DESCRIPTION
**Pull request by kelbyhawn**
Swaps out color variables from `shared/color.scss` with Primitive colors from `primitiveColors.css` on the Typography component. 

👀 **DOTD note:** there might be Eyes diffs with this.

## Links
Jira ticket: [DESIGN2-216](https://codedotorg.atlassian.net/browse/DESIGN2-216)

## Testing story
Tested locally in Storybook

----

## Before - using `shared/colors.scss`
Before

## After - using `primitiveColors.css`
After

### CODE REVIEW

File: apps/src/componentLibrary/typography/typography.module.scss [hunk 1]
```diff
@@ -87,7 +88,7 @@ html {
 */
 
 @mixin heading-common {
-  color: $neutral-dark;
+  color: var(--neutral-base-black);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/typography/typography.module.scss:91]: heading

File: apps/src/componentLibrary/typography/typography.module.scss [hunk 2]
```diff
@@ -136,7 +137,7 @@ html {
 // Paragraph styles
 @mixin paragraph-common {
   @include main-font-regular;
-  color: $neutral-dark;
+  color: var(--neutral-base-black);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/typography/typography.module.scss:140]: paragraph

**Comment by kelbyhawn** [comment]
@levadadenys I updated the PR title to be consistent with what you've been doing, and will do this for the remaining PRs too.

**Review by levadadenys — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**